### ISSUE TITLE
Logged out of all sites when Clef auth times out

### ISSUE THREAD

**Issue by josephwegner** [waltzio/waltz#27]
After the hour of being logged in, I get a chrome notification that I've been logged out of bypass.  I'm then logged out of all my sites.

**Comment by jessepollak** [comment]
Is this not the intended functionality?

**Comment by josephwegner** [comment]
Hm..  Well, I guess that's an interesting question.  And probably applies to Clef in general, not just Waltz.

I love being able to **choose** to log out from everywhere when I want to.  However, if I'm in the middle of working on a site, and the timer runs out, I do not want to be logged out.  I did not realize that was how Clef worked by default.

That seems pretty dangerous..  What if I was in the middle of entering data on some form, and all of a sudden the page refreshed and I was logged out?  That would really suck.

**Comment by jessepollak** [comment]
Yeah, figuring out the UX of this has been my biggest concern since I've had a day to play with it. I have a few thoughts:

1) On login, we could prompt the user whether they want to be logged out when the Clef session ends. This is annoying, but will put more prerogative on them.

2) On logout, we could give the user a 1 minute countdown where they have to reauth with Clef before being logged out of their sites. This isn't great from a security perspective (if a hacker had their computer and didn't already know Vault was installed, they would have 1 minute to figure out what to do). Part of this could be addressed by instant logouts if the Clef session is over and the browser reconnects to the internet and is opened. 

3) We could ignore the Clef logout.

4) We could have a setting in the options page, which dictates whether the user is logged out or not.

My favorite is (2), thoughts?

**Comment by josephwegner** [comment]
I think (2) is best, but we should add a setting to the options page so the user can turn off, per-domain, if they want to be automatically logged out.

**Comment by jessepollak** [comment]
Agreed.

**Comment by jessepollak** [comment]
To address this issue, we're going to add two things:

1) The ability to check the time remaining on a given access token (i.e. how much longer a user is logged in).

2) A web hook that triggers when there are 5 minutes remaining in a users session

@landakram and I will handle this on Clef's side sometime in the near future
